server/routes/reporting/standard-report/definitions.js

```diff
@@ -0,0 +1,121 @@
+
+(() => {
+  'use strict';
+
+  const { isCourtUser } = require('../../../components/auth/user-type');
+
+  // type IReportKey = {[key:string]: {
+  //   title: string,
+  //   apiKey: string,
+  //   search?: 'poolNumber' | 'dateRange', // etc only poolNumber is currently implemented
+  //   bespokeReport?: {
+  //     dao?: (req) => Promise<any>,                                 // custom data access function
+  //     insertColumns?: {[key: number]: [string, (data) => string]}, // column header, body
+  //   }
+  //   headings: string[], // corresponds to the ids provided for the headings in the API
+  //                       // (except report created dateTime)
+  // }};
+  module.exports.reportKeys = (app, req = null) => {
+    const courtUser = req ? isCourtUser(req) : false;
+
+    return {
+      'next-due': {
+        title: 'Next attendance date report',
+        apiKey: 'NextAttendanceDayReport',
+        search: 'poolNumber',
+        headings: [
+          'poolNumber',
+          'reportDate',
+          'poolType',
+          'reportTime',
+          'serviceStartDate',
+          'courtName',
+        ],
+      },
+      'undelivered': {
+        title: 'Undelivered list',
+        apiKey: 'UndeliverableListReport',
+        search: 'poolNumber',
+        headings: [
+          'poolNumber',
+          'reportDate',
+          'poolType',
+          'reportTime',
+          'serviceStartDate',
+          'courtName',
+          'totalUndelivered',
+        ],
+      },
+      'non-responded': {
+        title: 'Non-responded list',
+        apiKey: 'NonRespondedReport',
+        search: 'poolNumber',
+        headings: [
+          'poolNumber',
+          'reportDate',
+          'poolType',
+          'reportTime',
+          'serviceStartDate',
+          'courtName',
+          'totalNonResponded',
+        ],
+      },
+      'postponed-pool': {
+        title: 'Postponed list (by pool)',
+        apiKey: 'PostponedListByPoolReport',
+        search: 'poolNumber',
+        headings: [
+          'poolNumber',
+          'reportDate',
+          'poolType',
+          'reportTime',
+          'serviceStartDate',
+          'courtName',
+          'totalPostponed',
+        ],
+        searchUrl: app.namedRoutes.build('reports.postponed.search.get'),
+      },
+      'postponed-date': {
+        title: 'Postponed list (by date)',
+        apiKey: 'PostponedListByDateReport',
+        headings: [
+          'dateFrom',
+          'reportDate',
+          'dateTo',
+          'reportTime',
+          'totalPostponed',
+        ].concat(courtUser ? ['courtName'] : []),
+        grouped: {
+          headings: {
+            prefix: 'Pool ',
+            link: 'pool-overview',
+          },
+          totals: true,
+        },
+        searchUrl: app.namedRoutes.build('reports.postponed.search.get'),
+      },
+      'incomplete-service': {
```

Comment:
may want to add a searchUrl route back to the cutoff date page, that way your back link should work too.

Reply:
I've made a change to the renderer that checks for any reportType.searchUrl to use that, and uses the default one otherwise - this works here because I've hosted it on the default named path.